### ISSUE TITLE
DatePicker selects disabled date by default

### ISSUE THREAD

**Issue by ArneMangelschots** [formkit/formkit#1355]
### Reproduction

https://formkit.link/609623869f86fc952bc83da1f170dbaa

### Describe the bug

When i set a max-date on the DatePicker field the dates exceeding this max-date are disabled as expected. But whenever i open the DatePicker the date of today is selected and an error pops up. The date is disabled but looks selected, so this is some unexpected behaviour.

### Environment

• OS: [e.g. iOS]
• Browser [e.g. chrome, safari]
• Version [e.g. 22]

**Comment by fenilli** [comment]
It really does look like a bug for the second reason, I wouldn't see the "selected" as a bug, it is just showing the today's date ( might be good to change the visual style of it as to not be the same as selected ), but if you try to open the years it gives a lot of errors, and going to the correct date, and trying to open again it just simple stops working.

**Comment by ArneMangelschots** [comment]
The selected might not be a bug, but i do thinks it's some unexpected behaviour. I took a look at some other date pickers and the native ones, and they all set the initial state of the picker on the max-date if max-date is set.

**Comment by fenilli** [comment]
Should also be a good enhancement for clarity for the user, you could also do this now by setting value to -1 day of the max-date, but I agree that should be done automatically.

**Comment by ArneMangelschots** [comment]
Yes we thought about setting a default value to fix this problem, but we want the user to choose a date date himself and not being able to just proceed with the default date selected.

**Comment by ArneMangelschots** [comment]
@sashamilenkovic works great! thanks.
